--- mole.odevzdany	2023-04-27 16:17:36
+++ mole	2023-04-27 16:56:47
@@ -15,7 +15,7 @@
 		#write mole.sh,file_path,date,group to $MOLE_RC.
 		#write "mole.sh" or "$0" because getting $1 column for from $MOLE_RC is not stable. 
 		#$1 returns not only first column of the file string, but the path $MOLE_RC.
-	 	echo "$0,$fileName,$fileDir/$fileName,$(date +%Y-%m-%d),$groupFilter" >>"$MOLE_RC" 
+	 	echo "$0,$fileName,$fileDir/$fileName,$(date +%Y-%m-%d_%H-%M-%S),$groupFilter" >>"$MOLE_RC" 
 			
 		if [ -z  "$EDITOR" ]; then
 			if [ -z "$VISUAL" ]; then	
@@ -66,7 +66,7 @@
 	exit 1
 elif ! [ -f "$MOLE_RC" ]; then
 	mkdir -p "$(dirname "$MOLE_RC")" && touch "$MOLE_RC"
-	echo "MOLE_RC has been created"
+	#echo "MOLE_RC has been created"
 fi
 
 eval lastArg='$'$#
@@ -151,12 +151,12 @@
 
 		# Date ignored
 		if [ -n "$dateIgnored" ] && ! [ -d "$dateIgnored" ] && ! [ -f "$dateAfter" ]; then
-			fResult=$(echo "$fResult" | awk -v ignored="$dateIgnored" '{ if ($0 !~ ignored) print }')
+			fResult=$(echo "$fResult" | awk -v d="$dateIgnored" -F',' '{if ($4 >= d) print $0}')
 		fi 
 
 		# Date after
 		if [ -n "$dateAfter" ] && ! [ -d "$dateAfter" ] && ! [ -f "$dateAfter" ]; then
-			fResult=$(echo "$fResult" | awk -v d="$dateAfter" -F',' '{if ($4 >= d) print $0}')
+			fResult=$(echo "$fResult" | awk -v d="$dateAfter" -F',' '{if ($4 <= d) print $0}')
 		fi
 
 		# Realization of calling "mole list [FILTERS] [DIRECTORY]"
@@ -202,11 +202,11 @@
 			done
 			# Date ignored
 			if [ -n "$dateIgnored" ]; then
-				secretLogResult=$(echo "$secretLogResult" | awk -v ignrd="$dateIgnored" '{ if ($0 !~ ignrd) print }')
+				secretLogResult=$(echo "$secretLogResult" | awk -v a="$dateIgnored" -F',' '{if ($4 >= a) print $0}')
 			fi
 			# Date after
 			if [ -n "$dateAfter" ]; then
-				secretLogResult=$(echo "$secretLogResult" | awk -v a="$dateAfter" -F',' '{if ($4 >= a) print $0}')
+				secretLogResult=$(echo "$secretLogResult" | awk -v a="$dateAfter" -F',' '{if ($4 <= a) print $0}')
 			fi
 			
 			secretLogResult=$(echo "$secretLogResult" | awk 'NF' | awk -F',' '{print $3 ";" $4}' | sort)
@@ -226,13 +226,13 @@
 					secretLogDatesResult="$secretLogDatesResult$file;$dates\n"
 				fi
 			done
-			echo "$secretLogDatesResult" | grep -v '^$'| sort | bzip2 > "$secretLogPath"
-			echo "Secret log has been saved to: $secretLogPath"
+			echo "$secretLogDatesResult" | grep -v '^$'| sort -df | bzip2 > "$secretLogPath"
+			#echo "Secret log has been saved to: $secretLogPath"
 			exit
 		else
 
 		# Realization of calling "mole [-m] [FILTERS] [DIRECTORY]"
-		fResult=$(echo "$fResult" | awk -F',' '{print $2}'  | tail -r) 
+		fResult=$(echo "$fResult" | awk -F',' '{a[i++]=$2} END {for (j=i-1; j>=0;) print a[j--] }') 
 
 		# the most frequent file
 		if [ "$mostFrequent" = "true" ]; then
